testsrc/org/mozilla/javascript/tests/PullRequest857Test.java

```diff
@@ -0,0 +1,44 @@
+package org.mozilla.javascript.tests;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.NativeJavaObject;
+import org.mozilla.javascript.Scriptable;
+
+
+public class PullRequest857Test{

```

Comment:
I prefer the file name `JavaAdapterTest.java`. When we change the `JavaAdapter` in the future, it is easier to modify it if we know where the tests are already written.